chromium/background.js

```diff
@@ -88,26 +92,35 @@ loadStoredUserRules();
 
 /**
  * Set the icon color correctly
- * Depending on http-nowhere it should be red/default
+ * inactive: extension is enabled, but no rules were triggered on this page.
+ * blocking: extension is in "block all HTTP requests" mode.
+ * active: extension is enabled and rewrote URLs on this page.
+ * disabled: extension is disabled from the popup menu.
  */
-var setIconColor = function() {
-  var newIconPath = httpNowhereOn ? './icon38-red.png' : './icon38.png';
-  chrome.browserAction.setIcon({
-    path: newIconPath
+var updateState = function() {
+  chrome.tabs.query({active: true}, function(tabs) {
```

Comment:
You'll want to pass currentWindow:true here, too.

Otherwise, creation order determines the `tabs` array, and with multiple windows, multiple tabs can be active.

Example:
- Open a tab to a weird page (e.g. chrome://extensions)
- The extension icon will show as inactive
- Open a new window, and browse -- the icon state won't change